projects/RabbitMQ.Client/client/api/IConnectionFactory.cs

```diff
@@ -180,5 +180,15 @@ public interface IConnectionFactory
         /// timing out.
         /// </summary>
         TimeSpan ContinuationTimeout { get; set; }
+
+        /// <summary>
+        /// Set to a value greater than one to enable concurrent processing. For a concurrency greater than one <see cref="IBasicConsumer"/>
+        /// will be offloaded to the worker thread pool so it is important to choose the value for the concurrency wisely to avoid thread pool overloading.
+        /// <see cref="IAsyncBasicConsumer"/> can handle concurrency much more efficiently due to the non-blocking nature of the consumer.
+        /// Defaults to 1.
+        /// </summary>
+        /// <remarks>For concurrency greater than one this removes the guarantee that consumers handle messages in the order they receive them.
+        /// In addition to that consumers need to be thread/concurrency safe.</remarks>
+        int ProcessingConcurrency { get; set; }
```

Comment:
Can we call this `ConsumerDispatchConcurrency` or similar? "Processing" is not particularly specific.

Reply:
This name already passed all the reviews, is on master and also brought into `6.x`. What is causing the sudden name change request? Especially because I asked for this feedback as part of https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/866#discussion_r440790244

Btw. I'm not opposed to what you are proposing. Just wondering why

Reply:
The comment uses "consumer concurrency" which is roughly what I'm proposing. Things can easily be changed in master.

Reply:
I haven't participated in #866 and not every member of our team or community cares equally about naming. So as soon as I've noticed it I tried to suggest something clearer.

Reply:
Ok. I guess 6.2 isn't out yet so we could still fix it right?

Reply:
How about we merge this PR and then do the rename PR later so that we can also backport the rename?

Reply:
Yes, we can still change the name for `6.2`.

Reply:
Renaming can be done in another PR, no problem at all.